modules/gltf/gltf_document.cpp

```diff
@@ -5147,11 +5147,8 @@ GLTFMeshIndex GLTFDocument::_convert_mesh_to_gltf(Ref<GLTFState> p_state, MeshIn
 	TypedArray<Material> instance_materials;
 	for (int32_t surface_i = 0; surface_i < current_mesh->get_surface_count(); surface_i++) {
 		Ref<Material> mat = current_mesh->get_surface_material(surface_i);
-		if (p_mesh_instance->get_surface_override_material(surface_i).is_valid()) {
-			mat = p_mesh_instance->get_surface_override_material(surface_i);
-		}
-		if (p_mesh_instance->get_material_override().is_valid()) {
-			mat = p_mesh_instance->get_material_override();
+		if (p_mesh_instance->get_active_material(surface_i).is_valid()) {
+			mat = p_mesh_instance->get_active_material(surface_i);
 		}
```

Comment:
`get_active_material` already checks `surface_get_material`, so do we even need `get_surface_material`? I wonder if this would work:

```cpp
		Ref<Material> mat = p_mesh_instance->get_active_material(surface_i);
```

Reply:
yes. the one for CSG node doesn't have it let me check.

Reply:
I moved the check higher